tests/conftest.py

```diff
@@ -44,6 +46,349 @@ def pytest_logger_config(logger_config):
     logger_config.set_formatter_class(OCSLogFormatter)
 
 
+@pytest.fixture()
+def secret_factory(request):
+    """
+    Secret factory. Calling this fixture creates a new secret.
+    RBD based is default.
+    """
+    instances = []
+
+    def factory(interface=constants.CEPHBLOCKPOOL):
+        """
+        Args:
+            interface (str): CephBlockPool or CephFileSystem. This decides
+                whether a RBD based or CephFS resource is created.
+                RBD is default.
+        """
+        secret_obj = helpers.create_secret(
+            interface_type=interface
+        )
+        assert secret_obj, "Failed to create a secret"
+        instances.append(secret_obj)
+        return secret_obj
+
+    def finalizer():
+        """
+        Delete the RBD secrets
+        """
+        for instance in instances:
+            if not instance.is_deleted:
+                instance.delete()
+                instance.ocp.wait_for_delete(
+                    instance.name
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def block_pool_factory(request):
+    """
+    Create a Ceph block pool factory.
+    Calling this fixture creates new block pool instance.
+    """
+    instances = []
+
+    def factory():
+        cbp_obj = helpers.create_ceph_block_pool()
+        assert cbp_obj, "Failed to create block pool"
+        instances.append(cbp_obj)
+        return cbp_obj
```

Comment:
To make it more generic, we will need also the option to return the cephfilesystem object in case the interface is cephFileSystem

Reply:
It seems to me from code that we are currently not creating pool for CephFS but that we just look for it with [helpers.get_cephfs_data_pool_name()](https://github.com/red-hat-storage/ocs-ci/blob/cacbd94e4bfb05124ce1520d08478194d3c5684d/tests/helpers.py#L405). I am not sure if this is a good use case.
I would keep this factory as is because block pool is relevant only to RBD.

Reply:
We can change the name as well to make it more universal.
In case you prefer to keep it as is, let's merge this PR and I will make this more generic in different PR.
We really need the implementation of https://github.com/red-hat-storage/ocs-ci/pull/511 to enable the parametrize of RBD and cephFS